api/v1/checks.go

```diff
@@ -779,6 +779,23 @@ func (c GitHubCheck) GetEndpoint() string {
 	return strings.ReplaceAll(c.Query, " ", "-")
 }
 
+type GitLabCheck struct {
+	Description    `yaml:",inline" json:",inline"`
+	Templatable    `yaml:",inline" json:",inline"`
+	ConnectionName string       `yaml:"connection,omitempty" json:"connection,omitempty"`
+	Repo           string       `yaml:"repo" json:"repo"`
+	Username       string       `yaml:"username" json:"username"`
+	PasswordSecret types.EnvVar `yaml:"passwordSecret,omitempty" json:"passwordSecret"`
+}
+
+func (c GitLabCheck) GetType() string {
```

Comment:
It doesn't look like this check is specific to gitlab so we can probably rename it to `gitProtocol`

Reply:
Please check again - is this what you had in mind?

Reply:
Done